File: 001 交换两个数/main.swift
import Foundation

func swap1(num1: inout Int, num2: inout Int) {
    let temp = num1
    num1 = num2
    num2 = temp
}

func swap2(num1: inout Int, num2: inout Int) {
    num1 ^= num2
    num2 ^= num1
    num1 ^= num2
}

func swap3(num1: inout Int, num2: inout Int) {
    num1 = num1 + num2
    num2 = num1 - num2
    num1 = num1 - num2
}

func swap4(num1: inout Int, num2: inout Int) {
    num1 = num1 * num2
    num2 = num1 / num2
    num1 = num1 / num2
}

// 语言特性
func swap5(num1: inout Int, num2: inout Int) {
    (num1, num2) = (num2, num1)
}

var a = 10
var b = 20

swap4(num1: &a, num2: &b)

print(a, b)

File: 001 交换两个数/main.py
#-*- encoding:utf-8 -*-


def swap(num1, num2):
    temp = num1
    num1 = num2
    num2 = temp
    return num1, num2

def swap2(num1, num2):
    num1 = num1 + num2
    num2 = num1 - num2
    num1 = num1 - num2
    return num1, num2

def swap3(num1, num2):
    num1 = num1 * num2
    num2 = num1 / num2
    num1 = num1 / num2
    return num1, num2

def swap4(num1, num2):
    num1 ^= num2
    num2 ^= num1
    num1 ^= num2
    return num1, num2

def swap5(num1, num2):
    return num2, num1

if __name__ == "__main__":
    num1 = 10
    num2 = 20
    num1, num2 = swap5(num1, num2)
    
    print(num1, num2)